lib/enum/basic_enum.rb

```diff
@@ -22,48 +22,58 @@ def number_of_elements_in_collection
   # This one also uses 'count'
   # Try it yourself!
   def number_of_floats_or_fixnums
+    collection.select { |ele| ele.is_a? (Numeric)}.count
   end
 
+
   # Refer to: all?
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-all-3F
   def all_words_longer_than_length?(min_length)
+    collection.all? { |ele| ele.length > min_length }
   end
 
-
-  # For the next 3 methods refer to: collect or map
+  # For the next 3 methods refeall?r to: collect or map
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-any-3F
   def contains_a_word_longer_than?(min_length)
+    collection.any? { |ele| ele.length > min_length }
   end
 
   def capitalize_words
+    collection.map { |ele| ele.capitalize }
   end
 
   def square_numbers
+    collection.map { |ele| ele * ele}
   end
 
   # Use symbol to proc notation on this one!
   def reverse_words
+    collection.map { |ele| ele.reverse }
   end
 
   # Refer to collect_concat or flat_map for these next two
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-flat_map
   # To get the order right for the tests, you'll need to use the absolute value method
   def positive_and_negative_numbers
+    collection.flat_map { |ele| [ele.abs, -ele.abs] }
   end
 
   # Refer to detect or find
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-detect
   def find_first_awesome_person
+    collection.find {|hash| hash[:awesome] == true}
```

Comment:
Call your local variables by what they represent, not their datatype. So in this case, person instead of hash.